### PR TITLE
Menu info

### PR DESCRIPTION
**Pull request by lilyidi**

### CODE REVIEW

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    }
+
+    public LinkedList<TextView> addDiningHour(VenueInterval day, LinkedList<TextView> vertical){
+        TextView textView = new TextView(mActivity);
+        DateTimeFormatter intervalFormatter = DateTimeFormat.forPattern("yyyy-MM-dd");
+        DateTime dateTime = intervalFormatter.parseDateTime(day.date);
+        String dateString = dateTime.dayOfWeek().getAsText() + ", " + dateTime.monthOfYear().getAsString() + "/" + dateTime.dayOfMonth().getAsShortText();
+        textView.setText(dateString);
+        textView.setTextAppearance(mActivity, R.style.DiningInfoDate);
+        textView.setPadding(0, 10, 0, 10);
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: Should be incorporated in the `R.style.DiningInfoDate`

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                        public void call(List<Building> buildings) {
+                            if (!buildings.isEmpty()) {
+                                drawMarker(buildings.get(0).getLatLng());
+                            }
+                        }
+                    });
+        }
+    }
+
+    private void drawMarker(LatLng courseLatLng) {
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: `diningHallLatLng`

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        vertical.add(textView);
+        for (VenueInterval.MealInterval meal: day.meals){
+            TextView mealType = new TextView(mActivity);
+            mealType.setText(meal.type);
+            mealType.setId(vertical.getLast().getId() + 1);
+            mealType.setTextAppearance(mActivity, R.style.DiningInfoType);
+            RelativeLayout.LayoutParams layparammeal = new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.WRAP_CONTENT,
+                    RelativeLayout.LayoutParams.WRAP_CONTENT);
+            layparammeal.addRule(RelativeLayout.BELOW, vertical.getLast().getId());
+            layparammeal.setMargins(0, 10, 10, 0);
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: should be part of the XML styles

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            menuParent.addView(mealType, layparammeal);
+            vertical.add(mealType);
+
+            RelativeLayout.LayoutParams layparamtimes = new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.WRAP_CONTENT,
+                    RelativeLayout.LayoutParams.WRAP_CONTENT);
+            layparamtimes.addRule(RelativeLayout.ALIGN_PARENT_RIGHT, vertical.getLast().getId());
+            layparamtimes.addRule(RelativeLayout.ALIGN_BOTTOM, vertical.getLast().getId());
+            layparamtimes.setMargins(0, 10, 0, 0);
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: should be part of the XML styles

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        if (vertical.isEmpty()){
+            textView.setId(0);
+            textView.setId(textView.getId()+10);
+            menuParent.addView(textView);
+        } else {
+            textView.setId(vertical.getLast().getId()+1);
+            RelativeLayout.LayoutParams param = new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.WRAP_CONTENT,
+                    RelativeLayout.LayoutParams.WRAP_CONTENT);
+            param.addRule(RelativeLayout.BELOW, vertical.getLast().getId());
+            param.setMargins(0, 10, 10, 0);
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: Incorporate this into styles

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+        String dateString = dateTime.dayOfWeek().getAsText() + ", " + dateTime.monthOfYear().getAsString() + "/" + dateTime.dayOfMonth().getAsShortText();
+        textView.setText(dateString);
+        textView.setTextAppearance(mActivity, R.style.DiningInfoDate);
+        textView.setPadding(0, 10, 0, 10);
+        if (vertical.isEmpty()){
+            textView.setId(0);
+            textView.setId(textView.getId()+10);
+            menuParent.addView(textView);
+        } else {
+            textView.setId(vertical.getLast().getId()+1);
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: Add padding to the container itself

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java (hunk 3 above)
Comment (jasontsh) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: I'm not sure if that's possible as its directly creating the view @adelq

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/DiningHall.java [hunk 7]
```diff
@@ -18,17 +18,19 @@
 public class DiningHall implements Parcelable {
 
     private int id;
-    private String name;
+    private String name, location;
```
Comment (jasontsh) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/DiningHall.java]: unused field?

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java (hunk 3 above)
Comment (lilyidi) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/DiningInfoFragment.java]: @adelq @jasontsh I tried adding it to styles and it doesn't work.

**Comment by jasontsh** [comment]
LGTM but @adelq have a final look

**Comment by adelq** [comment]
LGTM I guess, will need refactoring at some point.

**Pull request merged**